app/queries/find_exact_match.rb

```diff
@@ -0,0 +1,39 @@
+class FindExactMatch
+  def initialize(query_field: :tags)
+    raise ArgumentError.new("query_field must be `:tags` or `:links`") unless %i{tags links}.include?(query_field)
+
+    @query_field = query_field.to_sym
+  end
+
+
+  def call(query_hash, document_type)
+    return [] unless query_hash.present?
+
+    subscriber_lists = subscriber_lists_with_all_matching_keys(query_hash)
```

Comment:
Appreciate this is an existing method, but `subscriber_lists_where_all_keys_match(query_hash)` reads better.